lib/config/db/couchdb-get-vendor.js

```diff
@@ -1,30 +1,37 @@
-module.exports = checkVendor
+module.exports = getVendor
 
-var findKey = require('lodash').findKey
+var defaults = require('lodash').defaults
 
 var toCouchDbUrl = require('../../utils/pouchdb-options-to-couchdb-url')
 var removeAuth = require('../../utils/remove-auth-from-url')
 
-function checkVendor (state, couch, callback) {
+function getVendor (state, couch, callback) {
   couch({url: '/'}, function (error, response, data) {
     if (error || (response && response.statusCode !== 200)) {
       var url = toCouchDbUrl(state.db.options)
       return callback(new Error('Could not find CouchDB at ' + removeAuth(url)))
     }
 
-    var vendor = findKey(data, function (property) {
-      return /^welcome/i.test(property)
+    var vendor = defaults(data.vendor, {
+      type: getVendorType(data.vendor)
     })
 
-    if (vendor !== 'couchdb') {
+    if (vendor.type  !== 'couchdb') {
       state.server.log(
         ['database', 'warn'],
         'You are not running an official CouchDB distribution, ' +
-        'but "' + vendor + '". ' +
+        'but "' + vendor.name + '". ' +
         'This might not be fully supported. Proceed at your own risk.'
       )
     }
 
-    callback(null)
+    callback(vendor)
   })
 }
+
+
+function getVendorType (vendor) {
+  if(vendor.name === 'The Apache Software Foundation') return 'couchdb'
```

Comment:
I’ve installed my CouchDB with `brew`, and `GET /` returns this for me:

```json
{"couchdb":"Welcome","uuid":"d0aabda74895f857e568407a31f830a5","version":"1.6.1","vendor":{"name":"Homebrew","version":"1.6.1_4"}}
```

Maybe @janl has a some idea on how to detect the vendor?
An alternative would be feature detection. For example we could `GET /_config` and if it returns `200`, then we know we can store config. Otherwise not. 

It reminds me of how we changed code based on browser agent string in the past, but eventually agreed on that feature detection is a better way to do it